src/Concordium/Client/GRPC2.hs

```diff
@@ -2359,42 +2360,343 @@ instance ToProto SendBlockItemInput where
+  where
+    msg = toProto bhInput
+
+-- |Get all fixed passive delegators for the reward period of a given block.
+-- In contrast to `getPassiveDelegatorsV2` which returns all delegators registered
+-- at the end of a given block, this returns all fixed delegators contributing
+-- stake in the reward period containing the given block.
+getPassiveDelegatorsRewardPeriodV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe (Seq.Seq QueryTypes.DelegatorRewardPeriodInfo)))
+getPassiveDelegatorsRewardPeriodV2 bhInput =
+    withServerStreamCollectV2 (callV2 @"getPassiveDelegatorsRewardPeriod") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = toProto bhInput
+
+
+-- |Get all registered passive delegators at the end of a given block.
+getPassiveDelegatorsV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe (Seq.Seq QueryTypes.DelegatorInfo)))
+getPassiveDelegatorsV2 bhInput =
+    withServerStreamCollectV2 (callV2 @"getPassiveDelegators") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = toProto bhInput
+
+-- |Get all fixed delegators of a given pool for the reward period of a given block.
+-- In contrast to `getPoolDelegatorsV2` which returns all active delegators registered
+-- for the given block, this returns all the active fixed delegators contributing stake
+-- in the reward period containing the given block.
+getPoolDelegatorsRewardPeriodV2 :: (MonadIO m) => BlockHashInput -> BakerId -> ClientMonad m (GRPCResult (Maybe (Seq.Seq QueryTypes.DelegatorRewardPeriodInfo)))
+getPoolDelegatorsRewardPeriodV2 bhInput baker =
+    withServerStreamCollectV2 (callV2 @"getPoolDelegatorsRewardPeriod") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto bhInput & ProtoFields.baker .~ toProto baker
+
+-- |Get all registered delegators of a given pool at the end of a given block.
+getPoolDelegatorsV2 :: (MonadIO m) => BlockHashInput -> BakerId -> ClientMonad m (GRPCResult (Maybe (Seq.Seq QueryTypes.DelegatorInfo)))
+getPoolDelegatorsV2 bhInput baker =
+    withServerStreamCollectV2 (callV2 @"getPoolDelegators") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto bhInput & ProtoFields.baker .~ toProto baker
+
+-- |Get IDs of all bakers at the end of a given block.
+getBakerListV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe (Seq.Seq BakerId)))
+getBakerListV2 bhInput = withServerStreamCollectV2 (callV2 @"getBakerList") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = toProto bhInput
+
+-- |Get key-value pairs representing the entire state of a specific contract instance in a given block.
+-- The resulting sequence is ordered lexicographically according to its keys.
+getInstanceStateV2 :: (MonadIO m) => BlockHashInput -> ContractAddress -> ClientMonad m (GRPCResult (Maybe (Seq.Seq (ByteString, ByteString))))
```

Comment:
Perhaps we should have a newtype or alias for `(ByteString, ByteString)`.

Reply:
Yes, I actually thought about that as well! However I think that reading the current type signature, it is clear that this is a key value pair. My concern is that by adding an alias/newtype, we simply add another level of abstraction and the user will have to look it up. But let me know what you think :-)

Reply:
Yea, you might be right.
I think you should be more explicit about what the return type is in the docs. Then it is fine.